bson/bsoncodec/struct_codec.go

```diff
@@ -97,7 +97,13 @@ func (sc *StructCodec) EncodeValue(r EncodeContext, vw bsonrw.ValueWriter, i int
 		}
 
 		ectx := EncodeContext{Registry: r.Registry, MinSize: desc.minSize}
-		err = encoder.EncodeValue(ectx, vw2, rv.Interface())
+
+		// HACK: pass *Document as is
```

Comment:
Why is this here? The `bson.Document` decoder should only be handling `**bson.Document`.